### PR TITLE
Allow to choose rounding type for discounts calculations

### PR DESCRIPTION
**Pull request by SzymJ**
This PR allows the choice of how prices should be rounded in discount calculations.

### CODE REVIEW

**Comment by artursmet** [comment]
@SzymJ Could you add a test case for this?

**Comment by SzymJ** [comment]
@artursmet tests added

File: tests/test_fractional_discount.py [hunk 1]
```diff
@@ -27,3 +27,20 @@ def test_discount_from_net():
...
+
+def test_precision_when_default_rounding():
+    """Test precision when default rounding is used.
+
+    Default rounding is set to ROUND_DOWN.
+    """
+    price = TaxedMoney(Money('1.01', 'BTC'), Money('1.01', 'BTC'))
+    result = fractional_discount(price, Decimal('0.5'))
+    assert result.net == Money('0.51', 'BTC')
+    assert result.net == Money('0.51', 'BTC')
```
Comment (artursmet) [inline tests/test_fractional_discount.py:40]: Why there are two assertions that check the same thing?

File: tests/test_fractional_discount.py [hunk 2]
```diff
@@ -27,3 +27,20 @@ def test_discount_from_net():
...
+
+def test_precision_when_half_up_rounding():
+    price = TaxedMoney(Money('1.01', 'BTC'), Money('1.01', 'BTC'))
+    result = fractional_discount(price, Decimal('0.5'), rounding=ROUND_HALF_UP)
+    assert result.net == Money('0.50', 'BTC')
+    assert result.net == Money('0.50', 'BTC')
```
Comment (artursmet) [inline tests/test_fractional_discount.py:46]: Same as above

File: tests/test_percentage_discount.py [hunk 3]
```diff
@@ -15,8 +16,18 @@ def test_discount():
...
     price = TaxedMoney(Money('1.01', 'BTC'), Money('1.01', 'BTC'))
     result = percentage_discount(price, percentage=50)
     assert result.net == Money('0.51', 'BTC')
     assert result.net == Money('0.51', 'BTC')
+
+def test_precision_when_half_up_rounding():
+    price = TaxedMoney(Money('1.01', 'BTC'), Money('1.01', 'BTC'))
+    result = percentage_discount(price, percentage=50, rounding=ROUND_HALF_UP)
+    assert result.net == Money('0.50', 'BTC')
+    assert result.net == Money('0.50', 'BTC')
```
Comment (artursmet) [inline tests/test_percentage_discount.py:33]: Another duplicated assert

**Review by artursmet — CHANGES_REQUESTED** [review]
Please clean up the assertions in tests

File: tests/test_percentage_discount.py (hunk 3 above)
Comment (SzymJ) [reply tests/test_percentage_discount.py]: Done

**Pull request merged**